airflow/jobs/backfill_job.py

```diff
@@ -183,7 +186,8 @@ def __init__(
         self.conf = conf
         self.rerun_failed_tasks = rerun_failed_tasks
         self.run_backwards = run_backwards
-        super().__init__(*args, **kwargs)
+        self.executor = executor or ExecutorLoader.get_default_executor()
+        super().__init__(executor=self.executor, *args, **kwargs)
```

Comment:
How about instead of doing this we make `executor` a lazy/cached property on base job?

That way it is not loaded until it is first accessed, which might fix the problem?